cfn-resources/database-user/cmd/resource/resource.go

```diff
@@ -258,41 +265,63 @@ func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler
 	}
+	log.Debugf("Update labels: %#+v", labels)
 
-	if currentModel.LdapAuthType == nil {
-		currentModel.LdapAuthType = new(string)
+	var scopes []mongodbatlas.Scope
+	for i, _ := range currentModel.Scopes {
+		s := currentModel.Scopes[i]
+		scope := mongodbatlas.Scope{
+			Name: *s.Name,
+			Type: *s.Type,
+		}
+		scopes = append(scopes, scope)
 	}
+	log.Debugf("Update scopes: %#+v", scopes)
+
 	groupID := *currentModel.ProjectId
-	username := *currentModel.Username
-	log.Debugf("groupID:%s, username:%s", groupID, username)
-	dbu := &mongodbatlas.DatabaseUser{
-		Roles:        roles,
-		GroupID:      groupID,
-		Username:     username,
-		Password:     *currentModel.Password,
-		DatabaseName: *currentModel.DatabaseName,
-		LDAPAuthType: *currentModel.LdapAuthType,
-		Labels:       labels,
+	log.Debugf("groupID: %#+v", groupID)
+
+	none := "NONE"
+	if currentModel.LdapAuthType == nil {
+		currentModel.LdapAuthType = &none
 	}
-	log.Debugf("dbu:%+v", dbu)
-	_, resp, err := client.DatabaseUsers.Update(context.Background(), groupID, username, dbu)
 
-	log.Debugf("Update resp:%+v", resp)
-	if err != nil {
-		log.Infof("Error Update database user:%s, error: %s", username, err)
-		if resp != nil && resp.StatusCode == 404 {
-			log.Warnf("Resource Not Found 404 for UPDATE groupId:%s, database user:%s, err:%+v, resp:%+v", groupID, username, err, resp)
+	if currentModel.AWSIAMType == nil {
+		currentModel.AWSIAMType = &none
+	}
+
+	if currentModel.X509Type == nil {
+		currentModel.X509Type = &none
+	}
+
+	if currentModel.Password == nil {
+		if (currentModel.LdapAuthType == &none) && (currentModel.AWSIAMType == &none) && (currentModel.X509Type == &none) {
+			err := fmt.Errorf("Password cannot be empty if not LDAP or IAM or X509: %v", currentModel)
 			return handler.ProgressEvent{
-				Message:          err.Error(),
 				OperationStatus:  handler.Failed,
-				HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
-		} else {
-			log.Warnf("Error UPDATE groupId:%s, database user:%s, err:%+v, resp:%+v", groupID, username, err, resp)
-			return handler.ProgressEvent{
 				Message:          err.Error(),
-				OperationStatus:  handler.Failed,
-				HandlerErrorCode: cloudformation.HandlerErrorCodeServiceInternalError}, nil
+				HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest}, nil
 		}
+		s := ""
+		currentModel.Password = &s
+	}
+
+	if (currentModel.X509Type != &none) || (currentModel.DeleteAfterDate == nil) {
+		s := ""
+		currentModel.DeleteAfterDate = &s
+	}
+
+	log.Printf("Check Delete after date here::???????")
+	spew.Dump(currentModel)
+
+	dbu := getDbUser(roles, groupID, currentModel, labels, scopes)
+	log.Debugf("dbu:%+v", dbu)
+	_, resp, err := client.DatabaseUsers.Update(context.Background(), groupID, *currentModel.Username, dbu)
+
+	log.Debugf("Update resp:%+v", resp)
+	if err != nil {
+		return progress_events.GetFailedEventByResponse(fmt.Sprintf("Error getting resource : %s", err.Error()),
```

Comment:
This logic is the same for the create. Can we extract it in a function and reuse it?